econplayground/api/views.py

```diff
@@ -6,13 +6,15 @@
 from rest_framework import viewsets
 from rest_framework.response import Response
 from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
 from econplayground.main.models import Graph, Submission
 from econplayground.api.serializers import (
     GraphSerializer, SubmissionSerializer
 )
 
 
 class GraphViewSet(viewsets.ModelViewSet):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
```

Comment:
Is there a need to remove this layer? I believe this prevents anonymous post/put access at the object level.

Reply:
Here I'm changing the permission from the app-wide `IsAuthenticated` specified in settings_shared to `IsAuthenticatedOrReadOnly`, because my intention is to allow the Graph module to be read by anonymous users. POST and PUT access is still prevented, as it has been - see `AnonymousGraphViewSetTest`